cmake/modules/AddSwift.cmake

```diff
@@ -589,6 +589,11 @@ function(add_swift_host_tool executable)
     set_target_properties(${executable} PROPERTIES
       BUILD_WITH_INSTALL_RPATH YES
       INSTALL_RPATH "@executable_path/../lib/swift/${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}")
+
+    get_target_triple(target target_variant "${SWIFT_HOST_VARIANT_SDK}" "${SWIFT_HOST_VARIANT_ARCH}"
+      MACCATALYST_BUILD_FLAVOR ""
+      DEPLOYMENT_VERSION "${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_DEPLOYMENT_VERSION}")
+    target_link_options(${executable} PRIVATE -target;${target})
```

Comment:
This needs to be protected like in the compile case: https://github.com/apple/swift/blob/master/cmake/modules/AddSwift.cmake#L85-L91

I think that the way to fix this is to invoke `_add_host_variant_c_compile_link_flags` in `add_swift_host_tool`.

Reply:
* I did not add the check since that a similar one I added in #31794 to `add_swift_host_library` was removed later in #32120
*  `_add_host_variant_c_compile_flags` is already called in `add_swift_host_tool `, but it does not add the `-target` flag to the link invocation (as expected after #31125 and #31612)
* not sure at this point if I should change instead `_add_host_variant_link_flags` (which I refrained to do in #31794 since from the code I skimmed it seemed we preferred to add flags specific to a platform outside of that)

Reply:
Discussed a bit offline, the gist is:

`_add_host_variant_c_compile_flags` - add flags for compilation
`_add_host_variant_c_compile_link_flags` - add flags for compilation and linking
`_add_host_variant_c_link_flags` - add flags for linking

This split originated in the idea that DRY is better, but increase the requirements for really understanding the custom CMake architecture.  In traditional CMake usage, this would not matter - you would specify the flags once as global flags and let CMake process them as appropriate.  Just calling the three each time is fine for now, and we should migrate towards merging them into a single call.

Reply:
Misread your comment -- I read `_add_host_variant_c_compile_link_flags` at first as `_add_host_variant_c_link_flags `...

Working on your suggestion

Reply:
This should be addressed now with f37dfc72a8a